sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalFileSystem.java

```diff
@@ -202,6 +202,10 @@ protected void delete(Collection<LocalResourceId> resourceIds) throws IOExceptio
 
   @Override
   protected LocalResourceId matchNewResource(String singleResourceSpec, boolean isDirectory) {
+    if (singleResourceSpec.endsWith(File.separator) && !isDirectory) {
```

Comment:
What if the character is escaped and being used in the name component of the path string?

Reply:
Is it even possible? For example according to https://stackoverflow.com/questions/9847288/is-it-possible-to-use-in-a-filename it's not. Or do you mean something different?

Reply:
Your right, it seems as though Linux doesn't allow `/`, Mac OS X doesn't allow `:`, Windows doesn't allow `\`.